kratos/geometries/nurbs_curve_on_surface_geometry.h

```diff
@@ -0,0 +1,294 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Thomas Oberbichler
+//                   Tobias Teschemacher
+//                   Andreas Apostolatos
+//
+//  Ported from the ANurbs library (https://github.com/oberbichler/ANurbs)
+//
+
+#if !defined(KRATOS_NURBS_CURVE_ON_SURFACE_H_INCLUDED )
+#define  KRATOS_NURBS_CURVE_ON_SURFACE_H_INCLUDED
+
+// Project includes
+#include "geometries/geometry.h"
+
+#include "geometries/nurbs_curve_geometry.h"
+#include "geometries/nurbs_surface_geometry.h"
+#include "geometries/nurbs_shape_function_utilities/nurbs_curve_shape_functions.h"
+#include "geometries/nurbs_shape_function_utilities/nurbs_interval.h"
+
+
+namespace Kratos {
+
+template <int TWorkingSpaceDimension, class TCurveContainerPointType, class TSurfaceContainerPointType>
+class NurbsCurveOnSurfaceGeometry : public Geometry<typename TCurveContainerPointType::value_type>
```

Comment:
I would go here for the nodes of the Surface. Because, essentially those are the one which are used as the dofs. Might be problematical if it is the other way round.

Reply:
Thanks for the input @tteschemacher. I am trying to get it fixed now, but I don't think I can create the necessary shared pointer only passing as argument,

TSurfaceContainerPointType const& ThisPoints

cause this class expects in its constructor a pointer to a surface and a pointer to a curve. Am I getting it right? The following code,

typename BaseType::Pointer Create(
        TSurfaceContainerPointType const& ThisPoints) const override
    {
        return Kratos::make_shared<NurbsCurveOnSurfaceGeometry>(ThisPoints);
    }

gives me the following compilation error,

error: no matching function for call to ‘Kratos::NurbsCurveOnSurfaceGeometry<3, Kratos::PointerVector<Kratos::Point>, Kratos::PointerVector<Kratos::Point> >::NurbsCurveOnSurfaceGeometry(const Kratos::PointerVector<Kratos::Point>&)’
  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }

Looking forward to some enlightening. Thanks!